app/core/models.py

```diff
@@ -103,7 +112,86 @@ class SearchSortOption(TimeStampedModel):
 
     def get_absolute_url(self):
         """ URL for displaying individual model records."""
-        return reverse('Configuration-detail', args=[str(self.id)])
+        return reverse('search-sort-option', args=[str(self.id)])
+
+    def __str__(self):
+        """String for representing the Model object."""
+        return f'{self.id}'
+
+
+class CourseDetailHighlight(TimeStampedModel):
+    """Model to contain course detail fields to display on results"""
+    HIGHLIGHT_ICON_CHOICES = [
+        ('clock', 'clock'),
+        ('hourglass', 'hourglass'),
+        ('user', 'user'),
+        ('multi_users', 'multi_users'),
+        ('location', 'location'),
+        ('calendar', 'calendar'),
+    ]
+
+    display_name = models.CharField(
+        max_length=200,
+        help_text='Enter the display name of the sorting option')
+    field_name = models.CharField(
+        max_length=200,
+        help_text='Enter the metadata field name as displayed in Elasticsearch'
+                  ' e.g. course.title'
+    )
+    xds_ui_configuration = models\
+        .ForeignKey(XDSUIConfiguration, on_delete=models.CASCADE,
+                    related_name='course_highlights')
+    active = models.BooleanField(default=True)
+    highlight_icon = models.CharField(
+        max_length=200,
+        choices=HIGHLIGHT_ICON_CHOICES,
+        default='user',
+    )
+    rank = \
+        models.IntegerField(default=1,
+                            help_text="Order in which highlight show on the "
+                                      "course detail page (2 items per row)",
+                            validators=[MinValueValidator(1,
+                                                          "rank shoud be at "
+                                                          "least 1")])
+
+    def get_absolute_url(self):
+        """ URL for displaying individual model records."""
+        return reverse('course-detail-highlight', args=[str(self.id)])
+
+    def __str__(self):
+        """String for representing the Model object."""
+        return f'{self.id}'
+
+    def save(self, *args, **kwargs):
+        num_active_highlights = \
+            CourseDetailHighlight.objects.filter(active=True).count()
+
+        # only 8 highlights can be active at any given time
+        if num_active_highlights >= 8:
```

Comment:
https://jira.deloitte.com/browse/ECC-647
One caveat is that we should still enforce a max because it affects the display if they pick a number that's too high. Thoughts?

Reply:
Is there any way to craft the display such that additional highlights could be located in a different spot?  It's fine to cap at 8 for now but ultimately we should have some way of accommodating as much flexibility as possible.  We can think about it more if/when the team picks up ECC-647, which will mostly likely be in a post-MVP project cycle.

Reply:
Makes sense - I will add this comment in the story so that we don't lose sight of it.